sql/core/src/main/scala/org/apache/spark/sql/Dataset.scala

```diff
@@ -576,6 +576,8 @@ class Dataset[T] private[sql](
     val parsedDelay =
       Option(CalendarInterval.fromString("interval " + delayThreshold))
         .getOrElse(throw new AnalysisException(s"Unable to parse time delay '$delayThreshold'"))
+    require(parsedDelay.milliseconds >= 0 && parsedDelay.months >= 0,
+      s"delay threshold ($delayThreshold) should not be negative.")
```

Comment:
use `require(parsedDelay.milliseconds >= 0 && parsedDelay.months >= 0)` to make `delayThreshold` more reasonable and significative.